app/controllers/application_controller.rb

```diff
@@ -1,28 +1,54 @@
 # encoding:UTF-8
 class ApplicationController < ActionController::Base
-  include TheRole::Controller
   protect_from_forgery
 
   before_filter :configure_permitted_devise_parameters, if: :devise_controller?
   before_filter :set_locale
 
+  rescue_from CanCan::AccessDenied do |ex|
+    flash[:error] = ex.message
+    render :text => '', :layout => true, :status => :forbidden
+  end
 
-  def access_denied
-    flash[:error] = t('the_role.access_denied')
-    redirect_to(:back)
-
-  rescue ActionController::RedirectBackError
-    redirect_to root_path
-  end 
+  rescue_from ActiveRecord::RecordInvalid do |ex|
+    flash[:error] = 
+      "Fel i formulär:  #{ex.record.errors.full_messages.join '; '}"
+    render referring_action, :status => :unprocessable_entity
+  end
 
   protected
-
   def configure_permitted_devise_parameters
     devise_parameter_sanitizer.for(:sign_in) {|u| u.permit(:username, :password, :remember_me)}
     devise_parameter_sanitizer.for(:sign_up) {|u| u.permit(:username, :email, :password, :password_confirmation) }
     devise_parameter_sanitizer.for(:account_update) { |u| u.permit(:password, :password_confirmation, :current_password) }
   end
 
+  def self.permission
+    self.name.gsub('Controller','').singularize.split('::').last.constantize.name rescue nil
+  end
+
+  def current_ability
+    @current_ability ||= Ability.new(current_user)
+  end
+
+  #load the permissions for the current user so that UI can be manipulated
+  def load_permissions(*args)
```

Comment:
Unused method argument - `args`. If it's necessary, use `_` or `_args` as an argument name to indicate that it won't be used. You can also write as `load_permissions(*)` if you want the method to accept any arguments but don't care about them.